### PR TITLE
Added Github actions

### PR DESCRIPTION
**Pull request by AbinavRavi**
## Description
Added github actions for linting and testing of pipeline-dp, [github actions issue](https://github.com/OpenMined/PipelineDP/issues/12)

## Checklist
- [x ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    branches: [ $default-branch ]
+
+jobs:
+  linting:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6,3.7, 3.8, 3.9]
```
Comment (chinmayshah99) [inline .github/workflows/tests.yml:16]: All versions of python are not needed for linting. One version of py version should work.

**Review by chinmayshah99 — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  test:
+    runs-on:  ${{ matrix.os }}
+    needs: [linting]
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.6,3.7, 3.8, 3.9]
```
Comment (dvadym) [inline .github/workflows/tests.yml:42]: Nit (style improvement): space is missing between 3.6 and 3.7

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+on:
+  push:
+    branches: [ $default-branch ]
+  pull_request:
+    branches: [ $default-branch ]
+
+jobs:
+  linting:
+
+    runs-on: ubuntu-latest
```
Comment (dvadym) [inline .github/workflows/tests.yml:12]: is it possible to run on Windows and Mac as well?

**Review by dvadym — COMMENTED** [review]
Thanks a lot for creation of Github actions. I left minor comments.

Also I have a question: does failing of linting and test jobs block push?

**Comment by chinmayshah99** [comment]
> Also I have a question: does failing of linting and test jobs block push?

Yes, they block push. It clearly shows test cases do not pass. Though we have administrator override.

File: .github/workflows/tests.yml (hunk 3 above)
Comment (chinmayshah99) [reply .github/workflows/tests.yml:12]: Linting tests are not required to be run on all OS. One should suffice. We can run test cases on all OS.

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        if [ -f requirements.dev.txt ]; then pip install -r requirements.dev.txt; fi
```
Comment (chinmayshah99) [inline .github/workflows/tests.yml:28]: are all packages of requirements.dev needed?

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        python -m pip install pylint --rcfile=pylintrc.dms pipeline_dp tests
+
+  test:
+    runs-on:  ${{ matrix.os }}
+    needs: [linting]
+    strategy:
+      fail-fast: false
+      max-parallel: 3
+      matrix:
+        os: [ubuntu-latest]
```
[the remark this answers is not in the document]
Comment (AbinavRavi) [reply .github/workflows/tests.yml]: I added it. It is available on the latest commit

**Review by dvadym — APPROVED** [review]
Thanks!

File: .github/workflows/tests.yml (hunk 4 above)
Comment (AbinavRavi) [reply .github/workflows/tests.yml:28]: Yes when I checked the tests they are using most requirements

**Review by chinmayshah99 — APPROVED** [review]

**Pull request merged**